pkg/registry/core/service/portallocator/controller/repair.go

```diff
@@ -112,14 +112,13 @@ func (c *Repair) doRunOnce() error {
 	// If etcd server is not running we should wait for some time and fail only then. This is particularly
 	// important when we start apiserver and etcd at the same time.
 	var snapshot *api.RangeAllocation
-
-	err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
-		var err error
-		snapshot, err = c.alloc.Get()
-		return err == nil, err
+	var lastError error
+	err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
why changing the interval from 1s to 2s?

Reply:
for not retry too often

Reply:
I have changed 1s